modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/NancyFXServerCodegen.java

```diff
@@ -0,0 +1,404 @@
+        addOption(PACKAGE_NAME, "C# package name (convention: Title.Case).", packageName);
+        addOption(PACKAGE_VERSION, "C# package version.", packageVersion);
+        addOption(SOURCE_FOLDER, SOURCE_FOLDER_DESC, sourceFolder);
+
+        // CLI Switches
+        addSwitch(SORT_PARAMS_BY_REQUIRED_FLAG, SORT_PARAMS_BY_REQUIRED_FLAG_DESC, sortParamsByRequiredFlag);
+        addSwitch(OPTIONAL_PROJECT_FILE, OPTIONAL_PROJECT_FILE_DESC, optionalProjectFileFlag);
+        addSwitch(USE_DATETIME_OFFSET, USE_DATETIME_OFFSET_DESC, useDateTimeOffsetFlag);
+        addSwitch(USE_COLLECTION, USE_COLLECTION_DESC, useCollection);
+        addSwitch(RETURN_ICOLLECTION, RETURN_ICOLLECTION_DESC, returnICollection);
+        addSwitch(IMMUTABLE_OPTION, "Enabled by default. If disabled generates model classes with setters", true);
+        typeMapping.putAll(nodaTimeTypesMappings());
+        languageSpecificPrimitives.addAll(nodaTimePrimitiveTypes());
+
+        importMapping.clear();
+    }
+
+    @Override
+    public CodegenType getTag() {
+        return SERVER;
+    }
+
+    @Override
+    public String getName() {
+        return "nancyfx";
+    }
+
+    @Override
+    public String getHelp() {
+        return "Generates a NancyFX Web API server.";
+    }
+
+    @Override
+    public void processOpts() {
+        super.processOpts();
+
+        apiPackage = isNullOrEmpty(packageName) ? API_NAMESPACE : packageName +  "." + API_NAMESPACE;
+        modelPackage = isNullOrEmpty(packageName) ? MODEL_NAMESPACE : packageName + "." + MODEL_NAMESPACE;
+
+        supportingFiles.add(new SupportingFile("parameters.mustache", sourceFile("Utils"), "Parameters.cs"));
+        supportingFiles.add(new SupportingFile("packages.config.mustache", sourceFolder(), "packages.config"));
+        supportingFiles.add(new SupportingFile("nuspec.mustache", sourceFolder(), packageName + ".nuspec"));
+
+        if (optionalProjectFileFlag) {
+            supportingFiles.add(new SupportingFile("Solution.mustache", "", packageName + ".sln"));
+            supportingFiles.add(new SupportingFile("Project.mustache", sourceFolder(), packageName + ".csproj"));
+        }
+        additionalProperties.put("packageGuid", packageGuid);
+
+        setupModelTemplate();
+        processImportedMappings();
+        appendDependencies();
+    }
+
+    private void setupModelTemplate() {
+        final Object immutableOption = additionalProperties.get(IMMUTABLE_OPTION);
+        if (immutableOption != null && "false".equalsIgnoreCase(immutableOption.toString())) {
+            log.info("Using mutable model template");
+            modelTemplateFiles.put("modelMutable.mustache", ".cs");
+        } else {
+            log.info("Using immutable model template");
+            modelTemplateFiles.put("model.mustache", ".cs");
+        }
+    }
+
+    private void processImportedMappings() {
+        for (final Entry<String, String> entry : ImmutableSet.copyOf(importMapping.entrySet())) {
```

Comment:
I think a comment here would be helpful, even if it's just an example of what's expected to be in entry prior to being processed.